spec/features/authentication/edit_password_spec.rb

```diff
@@ -41,4 +41,25 @@
 
     expect(page).to have_content "Sign out"
   end
+
+  it "stays on the same page and does not change the password, if the confirmation does not match" do
+    visit edit_user_registration_path
+
+    fill_in "Current password", with: user.password
+    fill_in "Password", with: pw
+    fill_in "Password confirmation", with: "different password"
+
+    click_on "Submit"
+
+    expect(page).to have_content "Password confirmation doesn't match Password"
+
+    sign_out
+
+    fill_in "Email", with: user.email
+    fill_in "Password", with: user.password
+
+    click_on "Continue"
+
+    expect(page).to have_content "Sign out"
```

Comment:
this implies the user successfully logs in?

Reply:
Yes - they are still using the original password, not the one they tried to change to (pw).

Reply:
OK I see. The logic is valid but inverted. Can you make the test hold the new password and make sure the new password does _not_ work?

Reply:
Fair enough - now changed